omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleActionResource1_8.java

```diff
@@ -0,0 +1,245 @@
+			}
+		}
+		
+		boolean needsRefresh = false;
+		for (Module module : moduleFactoryWrapper.getModulesInStartupOrder(modules)) {
+			if (moduleFactoryWrapper.isModuleStopped(module)) {
+				needsRefresh = moduleFactoryWrapper.startModuleSkipRefresh(module, servletContext) || needsRefresh;
+			}
+		}
+		if (needsRefresh) {
+			moduleFactoryWrapper.refreshWebApplicationContext(servletContext);
+		}
+	}
+	
+	private void unloadModules(Collection<Module> modules, ServletContext servletContext) {
+		boolean needsRefresh = false;
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStarted(module)) {
+				moduleFactoryWrapper.stopModuleSkipRefresh(module, servletContext);
+				needsRefresh = true;
+			}
+			moduleFactoryWrapper.unloadModule(module);
+		}
+		
+		if (needsRefresh) {
+			moduleFactoryWrapper.refreshWebApplicationContext(servletContext);
+		}
+	}
+	
+	private void stopModules(Collection<Module> modules, ServletContext servletContext) {
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStarted(module)) {
+				moduleFactoryWrapper.stopModule(module, servletContext);
+			}
+		}
+	}
+	
+	private void startModules(Collection<Module> modules, ServletContext servletContext) {
+		List<Exception> errors = new ArrayList<Exception>();
+		boolean needsRefresh = false;
+		if (modules.size() > 1) {
+			modules = moduleFactoryWrapper.getModulesInStartupOrder(modules);
+		}
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStopped(module)) {
+				module.clearStartupError();
+				needsRefresh = moduleFactoryWrapper.startModuleSkipRefresh(module, servletContext) || needsRefresh;
+				//module actions are executed in other thread, so we need to explicitly check and throw them
+				if (module.getStartupErrorMessage() != null) {
+					errors.add(new ModuleException(module.getStartupErrorMessage()));
+				}
+			}
+		}
+		//check if any module has been started, no sense to refresh WAC if all modules failed to start
+		if (needsRefresh) {
+			moduleFactoryWrapper.refreshWebApplicationContext(servletContext);
+		}
+		throwErrors(errors);
+	}
+	
+	private void throwErrors(List<Exception> errors) {
+		if (!errors.isEmpty()) {
+			StringBuilder stringBuilder = new StringBuilder();
+			for (Exception error : errors) {
+				stringBuilder.append(error.getMessage()).append("; ");
+			}
+			throw new RuntimeException(stringBuilder.toString());
+		}
+	}
+	
+	private Module findByUuid(String uuid) {
+		for (Module loadedModule : moduleFactoryWrapper.getLoadedModules()) {
+			if (ModuleResource1_8.getUuid(loadedModule).equals(uuid)) {
+				return loadedModule;
+			}
+		}
+		throw new IllegalArgumentException("There is no loaded module with uuid: " + uuid);
+	}
+	
+	@Override
+	public ModuleAction newDelegate() {
+		return new ModuleAction();
+	}
+	
+	@Override
+	public ModuleAction save(ModuleAction delegate) {
+		throw new IllegalArgumentException("ModuleAction cannot be saved");
```

Comment:
We should actually say here that since create() is overwritten this method must not be called ever... so probably UnsupportedOperationException would be better.